Messaging.md

```diff
@@ -87,33 +97,126 @@ The following DLC-specific types are used throughout the specification. All type
+#### Version 1 `contract_descriptor`
+
+1. type: 42784 (`contract_descriptor_v1`)
+2. data:
+   * [`u16`:`num_digits`]
+   * [`payout_function`:`payout_function`]
+   * [`rounding_intervals`:`rounding_intervals`]
+
+This type represents a numeric outcome contract.
+
+The type `rounding_intervals` is defined [here](NumericOutcome.md#rounding-interval-serialization).
 
 ### The `oracle_info` Type
 
-This type contains information about the oracle(s) to be used in executing a DLC, and possibly the outcomes possible if these are not specified in the corresponding `contract_info`.
+This type contains information about the oracles to be used in executing a DLC.
 
 #### Version 0 `oracle_info`
 
 1. type: 42770 (`oracle_info_v0`)
 2. data:
-   * [`x_point`:`oracle_public_key`]
-   * [`x_point`:`oracle_nonce`]
+   * [`oracle_announcement`:`oracle_announcement`]
+
+This type of oracle info is for single-oracle events.
+
+#### Version 1 `oracle_info`
+
+1. type: 42786 (`oracle_info_v1`)
+2. data:
+   * [`u16`:`threshold`]
+   * [`u16`:`num_oracles`]
+   * [`oracle_announcement`:`oracle_announcement_1`]
+   * ...
+   * [`oracle_announcment`:`oracle_announcement_num_oracles`]
+
+This type of oracle info is for multi-oracle events where all oracles are signing messages
+that exactly correspond to the messages being signed by the other oracles, and `threshold`
+oracles must be in agreement for execution to happen.
+
+#### Version 2 `oracle_info`
+
+1. type: 55340 (`oracle_info_v2`)
+2. data:
+   * [`u16`:`threshold`]
+   * [`u16`:`num_oracles`]
+   * [`oracle_announcement`:`oracle_announcement_1`]
+   * ...
+   * [`oracle_announcment`:`oracle_announcement_num_oracles`]
+   * [`oracle_params`:`oracle_params`]
+
+The order of the oracle announcements represents a total ordering of preference on the oracles.
+
+This type of oracle info is for multi-oracle numeric events where allowed differences in the values
+signed by oracles is specified in `oracle_params`.
+
+### The `oracle_params` Type
+
+contains information about how oracle information is used in a given contract.
+
+#### Version 0 `oracle_params` Type
+
+1. type: 55338 (`oracle_params_v0`)
+2. data
+   * [`u16`:`maxErrorExp`]
+   * [`u16`:`minFailExp`]
+   * [`bool`:`maximize_coverage`]
+
+This type is used when the error bound requirements for any set of oracles `threshold` oracles in a
+multi-oracle numeric outcome DLC with allowed error is the same.
+
+### The `negotiation_fields` Type
+
+This type contains preferences of the accepter of a DLC which are taken into account during DLC construction.
+
+#### Version 0 `negotiation_fields`
+
+1. type: 55334 (`negotiation_fields_v0`)
+2. data:
+   * (empty)
+
+This type signifies that the accepter has no negotiation fields.
+
+#### Version 1 `negotiation_fields`
+
+1. type: 55336 (`negotiation_fields_v1`)
+2. data
+   * [`rounding_intervals`: `rounding_intervals`]
```

Comment:
This is used both here and line 137 is that on purpose?

Reply:
Yes, you'll recall that both offerer and accepter get to give their max rounding and then these are merged before being used in CET computation.

Reply:
Oh right!